solver/llbsolver/vertex.go

```diff
@@ -322,13 +321,6 @@ func loadLLB(ctx context.Context, def *pb.Definition, polEngine SourcePolicyEval
 func llbOpName(pbOp *pb.Op, load func(digest.Digest) (solver.Vertex, error)) (string, error) {
 	switch op := pbOp.Op.(type) {
 	case *pb.Op_Source:
-		if id, err := source.FromLLB(op, nil); err == nil {
-			if id, ok := id.(*source.LocalIdentifier); ok {
-				if len(id.IncludePatterns) == 1 {
-					return op.Source.Identifier + " (" + id.IncludePatterns[0] + ")", nil
-				}
-			}
-		}
```

Comment:
Not sure what this was about, but it seemed pretty painful to pass a `*source.Manager` in here, and it didn't look load-bearing, so I removed it. :sweat_smile:

Reply:
Hm, this is definitely not particularly obvious what this is required for - I think it's just for naming the vertex, which is used in error messages, logs, traces, etc. It's used if `llb.WithCustomName` isn't set, so probably worth keeping it :smile: 

That said, maybe a neater implementation wouldn't be specific to the `LocalIdentifier` - maybe we could just have a formatted version of the schema and the attributes map? Ideally, we could have each `Identifier` implement a `.Name` method, but then as you say, we'd need to pass in a `*source.Manager` which, while doable, feels weird to do, just to get a name out :thinking:

Reply:
Personally I was surprised to even see this here - it seems like it could be stale code. There are so many other types of Source identifiers; why does it specialize for Local, and only when there's one IncludePattern? :thinking: 

It looks like it also predates `WithCustomName`:

* https://github.com/moby/buildkit/commit/4c44d2c6e41fcebe424e522322f9dea8edebc195 <- Jul 31, 2018
* https://github.com/moby/buildkit/commit/1805895e5843f758dd64a16c58c7c205071f2b0d <- Jan 31, 2018

Considering that, my vote would be to just remove it.

Reply:
> * [4c44d2c](https://github.com/moby/buildkit/commit/4c44d2c6e41fcebe424e522322f9dea8edebc195) <- Jul 31, 2018
> 
> * [1805895](https://github.com/moby/buildkit/commit/1805895e5843f758dd64a16c58c7c205071f2b0d) <- Jan 31, 2018

Good catch it seems to be an oversight :+1: